sn_transfers/src/wallet/data_payments.rs

```diff
@@ -163,4 +163,67 @@ impl PaymentQuote {
             signature: vec![],
         }
     }
+
+    /// Check whether self is newer than the target quote.
+    pub fn is_newer_than(&self, other: &Self) -> bool {
+        self.timestamp > other.timestamp
+    }
+
+    /// Check against a new quote, verify whether it is a valid one from self perspective.
+    /// Returns `true` to flag the `other` quote is valid, from self perspective.
+    pub fn historical_verify(&self, other: &Self) -> bool {
```

Comment:
Can we get some unit tests here to clearly lay out each of the error / fake quote cases?

Reply:
ok, will do